ddl/index.go

```diff
@@ -483,6 +483,11 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 		indexInfo.Unique = unique
 		indexInfo.ID = allocateIndexID(tblInfo)
 		tblInfo.Indices = append(tblInfo.Indices, indexInfo)
+
+		if err := checkInvisibleIndexOnPK(tblInfo); err != nil {
+			job.State = model.JobStateCancelled
```

Comment:
No test case for it?

Reply:
tested in https://github.com/pingcap/tidb/pull/17151/files#diff-703ae6b7872b425273d1832c198598c8R2100,
`alter table add index` is same with `create index on table`